source/1_engine/rex_engine/src/string/stringid.cpp

```diff
@@ -0,0 +1,137 @@
+#include "rex_engine/string/stringid.h"
+
+#include "rex_engine/string/stringpool.h"
+#include "rex_std/memory.h"
+#include "rex_std/vector.h"
+
+namespace rex
+{
+  namespace internal
+  {
+    //-------------------------------------------------------------------------
+    StringEntryID make(rsl::string_view characters)
+    {
+      if(characters.empty())
+      {
+        return StringEntryID(); // StringID::is_none() == true
+      }
+
+      return *string_pool::store(characters);
+    }
+  } // namespace internal
+
+  //-------------------------------------------------------------------------
+  /**
+   * Create an empty StringID.
+   */
+  StringID::StringID()
+      : m_comparison_index(0)
```

Comment:
can this use an internal invalid id?
it's hashing so it's never be 0 afaik but a proper state that this is an invalid id is probably useful here

Reply:
It's not an invalid state, it's the none state. meaning this is empty.

Reply:
what exactly is an empty string id definition?
a string id is an id that identifies a string, right?
in this case, it doesn't identify any string, which is a special case.
if that's a valid state or not doesn't really matter, I still think a proper named value to assign here is worth it

Reply:
The 0 is a hash to "Invalid String" so if you would call toString on an empty ( aka 0 is stored inside the comparison index ) you retrieve "Invalid String" so in theory it's not empty. I can store some special value if you want but is 0 not good enough?

Reply:
0 is fine, I just want it behind a name